status/controller.go

```diff
@@ -78,21 +88,41 @@ func (c *Controller[T]) Reconcile(ctx context.Context, req reconcile.Request) (r
 		ConditionCount.With(prometheus.Labels{
 			MetricLabelGroup:           gvk.Group,
 			MetricLabelKind:            gvk.Kind,
-			MetricLabelNamespace:       string(req.Namespace),
-			MetricLabelName:            string(req.Name),
-			MetricLabelConditionType:   string(condition.Type),
+			MetricLabelNamespace:       req.Namespace,
+			MetricLabelName:            req.Name,
+			MetricLabelConditionType:   condition.Type,
 			MetricLabelConditionStatus: string(condition.Status),
+			MetricLabelConditionReason: condition.Reason,
 		}).Set(1)
+		ConditionCurrentStatusSeconds.With(prometheus.Labels{
+			MetricLabelGroup:           gvk.Group,
+			MetricLabelKind:            gvk.Kind,
+			MetricLabelNamespace:       req.Namespace,
+			MetricLabelName:            req.Name,
+			MetricLabelConditionType:   condition.Type,
+			MetricLabelConditionStatus: string(condition.Status),
+			MetricLabelConditionReason: condition.Reason,
+		}).Set(time.Since(condition.LastTransitionTime.Time).Seconds())
```

Comment:
Any reason we don't want milliseconds here? Just too granular?

Reply:
Yeah, I think milliseconds is too granular. Given that monitoring will most likely be on something that is stuck for an extended period of time (many, many seconds), I think seconds makes the most sense.